packages/bull-vault/test/e2e/BullStrategyTestFork.t.sol

```diff
@@ -1,43 +1,179 @@
+        bullStrategy.deposit{value: wethToLend}(_crabAmount);
+        vm.stopPrank();
+        
+        assertEq(bullStrategy.balanceOf(user1).sub(userBullBalanceBefore), bullToMint);
+        assertTrue(wethToLend.sub(IEulerEToken(eToken).balanceOfUnderlying(address(bullStrategy)).sub(ethInLendingBefore)) <= 2);
+        assertEq(IEulerDToken(dToken).balanceOf(address(bullStrategy)).sub(usdcBorrowedBefore), usdcToBorrow);
+        assertEq(IERC20(usdc).balanceOf(user1).sub(userUsdcBalanceBefore), usdcToBorrow);
+    }
+
+    function _getCrabVaultDetails() internal view returns (uint256, uint256) {
+        VaultLib.Vault memory strategyVault = IController(address(controller)).vaults(crabV2.vaultId());
+
+        return (strategyVault.collateralAmount, strategyVault.shortAmount);
+    }
+
+    function _calcBullToMint(uint256 _crabToDeposit) internal view returns (uint256) {
+        if (IERC20(bullStrategy).totalSupply() == 0) {
+            return _crabToDeposit;
+        } else {
+            uint256 share = _crabToDeposit.wdiv(IERC20(crabV2).balanceOf(address(bullStrategy)));
+            uint256 bullTotalSupply = bullStrategy.totalSupply();
+            return share.wmul(bullTotalSupply).wdiv(uint256(1e18).sub(bullTotalSupply));
+        }     
+    }
+
+    function _calcCollateralAndBorrowAmount(uint256 _crabToDeposit) internal view returns (uint256, uint256) {
+        uint256 wethToLend;
+        uint256 usdcToBorrow;
+        if (IERC20(bullStrategy).totalSupply() == 0) {
+            {
+                uint256 ethUsdPrice = uniBullHelper.getTwap(
+                    controller.ethQuoteCurrencyPool(), controller.weth(), controller.quoteCurrency(), TWAP, false
+                );
+                uint256 squeethEthPrice =
+                    uniBullHelper.getTwap(controller.wPowerPerpPool(), controller.wPowerPerp(), controller.weth(), TWAP, false);
+                (uint256 ethInCrab, uint256 squeethInCrab) = _getCrabVaultDetails();
+                uint256 crabUsdPrice = (ethInCrab.wmul(ethUsdPrice).sub(squeethInCrab.wmul(squeethEthPrice).wmul(ethUsdPrice)))
+                    .wdiv(crabV2.totalSupply());
+                wethToLend = bullStrategy.TARGET_CR().wmul(_crabToDeposit).wmul(crabUsdPrice).wdiv(ethUsdPrice);
+                usdcToBorrow = wethToLend.wmul(ethUsdPrice).wdiv(bullStrategy.TARGET_CR()).div(1e12);
+            }
+        }
+        else {
+            uint256 share = _crabToDeposit.wdiv(IERC20(crabV2).balanceOf(address(bullStrategy)));
+            wethToLend = IEulerEToken(eToken).balanceOfUnderlying(address(bullStrategy)).wmul(share).wdiv(uint256(1e18).sub(share));
+            usdcToBorrow = IEulerDToken(dToken).balanceOf(address(bullStrategy)).wmul(share).wdiv(uint256(1e18).sub(share)).div(1e12);
+        }
+
+        return (wethToLend, usdcToBorrow);
+    }
+
+    function _calcTotalEthDelta(uint256 _crabToDeposit) internal view returns (uint256) {
```

Comment:
I don't understand what this function does, and not sure its correct for whatever its trying to do.  Hard to know as don't see it called anywhere yet.

Reply:
Yeah I see it is not used, I thought when started writing test I may need a function to get the bull delta